### PR TITLE
Intel asciidoc extensions

### PR DESCRIPTION
**Pull request by bashbaug**
This pull request adds the asciidoc spec source for several Intel OpenCL extensions:

* cl_intel_subgroups
* cl_intel_subgroups_short
* cl_intel_required_subgroup_size
* cl_intel_spirv_subgroups
* cl_intel_spirv_media_block_io
* cl_intel_spirv_device_side_avc_motion_estimation

### CODE REVIEW

**Comment by oddhack** [comment]
Oh, never mind my email comment asking where the sources were. BTW I think .adoc is more customary than .asciidoc.

**Comment by alycm** [comment]
Is it intended that vendor extensions should also be added to extensions,txt?

File: extensions/Makefile [hunk 1]
```diff
@@ -19,7 +19,7 @@
 # - pdf
 #
 
-all: html pdf
+all: html
```
[the remark this answers is not in the document]
Comment (bashbaug) [reply extensions/Makefile]: Fixed in 23839bf.

**Comment by bashbaug** [comment]
> Is it intended that vendor extensions should also be added to extensions,txt?

I initially envisioned extensions.txt as a way to create a single doc with all vendor extensions enabled, but this isn't something that's currently published or even built by default.  Is there a need for this capability? 
 If not, perhaps it should be removed, for now at least.

**Comment by alycm** [comment]
> I initially envisioned extensions.txt as a way to create a single doc with all vendor extensions enabled, but this isn't something that's currently published or even built by default. Is there a need for this capability?
If not, perhaps it should be removed, for now at least.

I can see that having a single document with all vendor extensions could be nice, the vendor extensions added in this PR have not been added to extensions.txt though.  Also, the comments in extensions.txt talk about khr etc extensions, not vendor extensions.

However, my original question was really just intended as "should these vendor extensions be added to extensions.txt in this PR?", and in practice it seems like there is no reason to do that currently.

**Comment by bashbaug** [comment]
I went ahead and added the Intel extensions to extensions.txt just in case we decide to resurrect this capability.  I also refactored the doc slightly so the comments are more clear what it is intended to do.

The HTML output looks pretty good with these changes.  The PDF output will need some work though (the table of contents has too many levels and some of the table text renders poorly, among other things).

I'd like this PR to be accepted soon if there are no other issues - when I uploaded the HTML output to the registry for the subgroup extensions I also uploaded a new txt document linking to the HTML output and to the asciidoc source, and the asciidoc source link will be broken until this PR is accepted.  Thanks!

**Comment by alycm** [comment]
LGTM.

**Pull request merged**